modules/constructor/constructor-basic/constructor-basic.services.js

```diff
@@ -0,0 +1,130 @@
+const ConstructorBasic = require('./constructor-basic.model');
+const Currency = require('../../currency/currency.model');
+const { deleteFiles, uploadFiles } = require('../../upload/upload.service');
+const {
+  BASIC_NOT_FOUND,
+  BASIC_ALREADY_EXIST,
+  IMAGE_NOT_PROVIDED,
+} = require('../../../error-messages/constructor-basic-messages');
+
+class ConstructorBasicService {
+  async getAllBasics({ skip, limit }) {
+    const items = await ConstructorBasic.find().populate(
+      'material',
+    )
+      .skip(skip)
+      .limit(limit);
+
+    const count = await ConstructorBasic.find().countDocuments();
+    return {
+      items,
+      count,
+    };
+  }
+
+  async getBasicById(id) {
+    const foundBasic = await ConstructorBasic.findById(id).populate(
+      'material',
+    );
+    if (foundBasic) {
+      return foundBasic;
+    }
+    throw new Error(BASIC_NOT_FOUND);
+  }
+
+  async calculatePrice(price) {
```

Comment:
Please move this method to separate currency file.

Reply:
Fixed